Builder/Admin/ListBuilder.php

```diff
@@ -49,7 +49,11 @@ public function getFilterColumns()
 
     protected function findFilterColumns()
     {
-        foreach ($this->getAllFields() as $columnName) {
+        $columns = $this->getVariable('filters');
+        if (null == $columns || 0 == sizeof($columns)) {
+            $columns = $this->getDisplayColumns();
```

Comment:
If we want to be able to filter on non displayed fields, you should better call `getAllFields()`

Could you please update this? And so we should be able to merge it.

Thank you!